### PR TITLE
Use gsha256sum when sha256sum doesn't exist [ Re: #1584 ]

### PR DESCRIPTION
**Pull request by fabianhjr**

### CODE REVIEW

File: src/Core/Options.idr [hunk 1]
```diff
@@ -229,11 +231,21 @@ defaultElab : ElabDirectives
...
-                     defaultElab Nothing Nothing
-                     (MkPrimNs Nothing Nothing Nothing Nothing) []
-                     []
+defaultHashFn : Core String
+defaultHashFn
+    = do Nothing <- coreLift $ pathLookup ["sha256sum", "gsha256sum", "sha256"]
+           | Just p => pure p
+         Nothing <- coreLift $ pathLookup ["openssl"]
+           | Just p => pure "openssl sha256 -r"
+         pure "sha256sum"
```
Comment (gallais) [inline src/Core/Options.idr:240]: Should this be an exception? We've already attempted pathLookup on `"sha256sum"`
so a priori it does not exist and returning it has no chance of success?

Comment (fabianhjr) [reply src/Core/Options.idr:240]: Yeah, it should be an exception.

Comment (fabianhjr) [reply src/Core/Options.idr:240]: Amending

**Comment by benhormann** [comment]
@fabianhjr BSD's `sha256` requires `-r` (except NetBSD has no such option - [man page](https://man.netbsd.org/sha256.1)).

Why not opt for BSD style output instead?
Use coreutil's `--tag` and drop `-r` for openssl. Change [src/Libraries/Utils/Binary.idr#L484](https://github.com/idris-lang/Idris2/blob/bc9ff537248500b143d2b7da479f1e51a0f2d500/src/Libraries/Utils/Binary.idr#L484) to grab the last word.

File: src/Core/Options.idr (hunk 1 above)
Comment (fabianhjr) [reply src/Core/Options.idr]: Amended for an error explaining the attempts.

**Comment by fabianhjr** [comment]
There was a runner issue on macOS CI, rerunning should clear the issue.

**Comment by fabianhjr** [comment]
Rebased for the runner re-run @gallais.

**Pull request merged**